Source/RunActivity/Viewer3D/WebServices/SwitchPanel/SwitchOnPanelDefinition.cs

```diff
@@ -0,0 +1,336 @@
+                    break;
+                case ORTS.Common.Input.UserCommand.ControlCircuitBreakerClosingOrder:
+                    description = Viewer.Catalog.GetParticularString("SwitchPanel", "Circuit Breaker");
+                    break;
+                case ORTS.Common.Input.UserCommand.ControlTractionCutOffRelayClosingOrder:
+                    description = Viewer.Catalog.GetParticularString("SwitchPanel", "Traction Cut-Off");
+                    break;
+                case ORTS.Common.Input.UserCommand.ControlDieselPlayer:
+                    description = Viewer.Catalog.GetParticularString("SwitchPanel", "Diesel Player");
+                    break;
+                case ORTS.Common.Input.UserCommand.ControlDieselHelper:
+                    description = Viewer.Catalog.GetParticularString("SwitchPanel", "Diesel Helper");
+                    break;
+                case ORTS.Common.Input.UserCommand.ControlHeadlightIncrease:
+                    description = Viewer.Catalog.GetParticularString("SwitchPanel", "Front light");
+                    break;
+                case ORTS.Common.Input.UserCommand.ControlCylinderCocks:
+                    description = Viewer.Catalog.GetParticularString("SwitchPanel", "Cyl Cocks");
+                    break;
+                case ORTS.Common.Input.UserCommand.ControlRetainersOn:
+                    description = Viewer.Catalog.GetParticularString("SwitchPanel", "Retainers");
+                    break;
+            }
+
+            return description;
+        }
+
+        public override bool Equals(object obj)
+        {
+            //Check for null and compare run-time types.
+            if ((obj == null) || !this.GetType().Equals(obj.GetType()))
+            {
+                return false;
+            }
+
+            if (((SwitchOnPanelDefinition)obj).NoOffButtons != this.NoOffButtons)
+            {
+                return false;
+            }
+
+            if (NoOffButtons > 0)
+            {
+                for (int i = 0; i < NoOffButtons; i++)
+                {
+                    if (((SwitchOnPanelDefinition)obj).UserCommand[i] != UserCommand[i])
+                    {
+                        return false;
+                    }
+                }
+            }
+
+            return ((SwitchOnPanelDefinition)obj).Description == Description;
+        }
+
+        public static void deepCopy(SwitchOnPanelDefinition to, SwitchOnPanelDefinition from)
+        {
+            to.NoOffButtons = from.NoOffButtons;
+            to.Description = from.Description;
+
+            to.UserCommand = new UserCommand[from.NoOffButtons];
+            for (int i = 0; i < from.NoOffButtons; i++)
+            {
+                to.UserCommand[i] = from.UserCommand[i];
+            }
+        }
+
+        public override int GetHashCode()
+        {
+            var hashCode = 1410623761;
```

Comment:
I'm curious, how do you set those numbers?

Reply:
Without this function I get the VS Compiler Warning (level 3) CS0659:
'class' overrides Object.Equals(object o) but does not override Object.GetHashCode()
To get rid of this warning you can have VS generate this function. If doing so, it generates another number every time. 
This function ObjectGetHashCode is not that important. But for new sources I try to suppress every warning.

Reply:
Thanks for the explanation!